client/tests/e2e/DataManagement.e2e.test.js

```diff
@@ -12,10 +12,15 @@ describe('Data Management common traits', () => {
   it('renders page with introduction', async () => {
     await getPage({ role: false, url: '/data-management' }, async page => {
       const introductionHeadingSelector = 'h2 ::-p-text(Introduction)';
-      const introductionHeadingText = await text(
-        page,
-        introductionHeadingSelector
-      );
+      const errors = await checkConsoleErrors(page, async () => {
```

Comment:
This is great but I think it has a chance to miss errors on certain element checks or with how the page element is being loaded. In this instance, errors could happen after the "Introduction" text has loaded which is what I'm guessing is happening in the following screenshot. There is a prop error not being caught on the DataManagement page:

![screenshot showing error with prop being passed to DataManagementRow](https://github.com/user-attachments/assets/971a7398-defe-430f-ad98-512760d7a071)

I'm thinking the `page.on('console')` listener could be done in `getPage` and passed into the callback the same way the browser and baseUrl and checked in that way